src/main/java/ast/interfaces/NodeBuilder.java

```diff
@@ -0,0 +1,46 @@
+package ast.interfaces;
```

Comment:
I renamed the `NodeFactory` class to `NodeBuilder` to give it a name more consistent with what it does (building these AST nodes through multiple steps)